apps/omg_eth/lib/omg_eth/root_chain/submit_block.ex

```diff
@@ -51,18 +51,18 @@ defmodule OMG.Eth.RootChain.SubmitBlock do
   @spec contract_transact(atom(), address, address, binary, [any], keyword) :: {:ok, hash()} | {:error, any}
   defp contract_transact(:infura = backend, _from, to, signature, args, opts) do
     abi_encoded_data = encode_tx_data(signature, args)
-    [nonce: nonce, gasPrice: gas_price, value: 0, gas: 100_000] = opts
+    [nonce: nonce, gasPrice: gas_price, value: value, gas: gas_limit] = opts
```

Comment:
Is the opts ordering guaranteed? E.g.

```elixir
iex(1)> [nonce: nonce, gasPrice: gas_price, value: value, gas: gas_limit] = [nonce: 1234, gasPrice: 10, value: 0, gas: 100_000]
[nonce: 1234, gasPrice: 10, value: 0, gas: 100000]

iex(2)> [nonce: nonce, gasPrice: gas_price, value: value, gas: gas_limit] = [nonce: 1234, gasPrice: 10, gas: 100_000, value: 0]
** (MatchError) no match of right hand side value: [nonce: 1234, gasPrice: 10, gas: 100000, value: 0]
```

Reply:
yeah in this case `contract_transact` is a private function and `opts` is set in https://github.com/omisego/elixir-omg/pull/1049/files#diff-b4e49282395f144255163b79a7368e66L44